web/packages/teleport/src/Roles/Roles.tsx

```diff
@@ -48,15 +48,28 @@ import { RoleList } from './RoleList';
 import templates from './templates';
 import { State, useRoles } from './useRoles';
 
-export function RolesContainer() {
+// RoleDiffProps are an optonal set of props to render the role diff visualizer.
+// This type union will ensure that either all fields are undefined or all are supplied
+type RoleDiffProps =
+  | { RoleDiffComponent?: undefined; updateRoleDiff?: undefined }
```

Comment:
If `roleDiffProps` is optional, why would I pass `rollDiffProps={{}}` (which matches this type)?

Reply:
yeah i think i might have overthought this one. updated to just make `roleDiffProps` optional and typed the `RoleDiffProps` normally